src/plugins/data/common/osd_field_types/osd_field_types.ts

```diff
@@ -84,3 +84,19 @@ export const setOsdFieldOverrides = (newOverrides: { [key: string]: any } | unde
 export const getOsdFieldOverrides = (): { [key: string]: any } => {
   return osdFieldOverrides;
 };
+
+/**
+ *  Mapping function from SQL_TYPES to OSD_FIELD_TYPES
+ *
+ *  @param {string} sqlType
+ *  @return {OSD_FIELD_TYPES}
+ */
+export function castSQLTypeToOSDFieldType(
```

Comment:
Agree, will move this out of `osd_fields` altogether. This seems to be conflicting with earlier review from Rocky https://github.com/opensearch-project/OpenSearch-Dashboards/pull/8337#discussion_r1776826366. In future, we can define clear tenets so that the contributors know the clear delineation between what goes in data plugin and what goes in query enhancement.

Reply:
Updated here: [6d6a9c6](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/8609/commits/6d6a9c6b6100e4ca696875e237fe793c07fd84a8)